pkg/kafka/auth/scram.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"crypto/sha256"
+	"crypto/sha512"
+	"hash"
+
+	"github.com/Shopify/sarama"
+	"github.com/pkg/errors"
+	SCRAM "github.com/xdg/scram"
+)
+
+// SCRAMConfig describes the configuration properties required for the SCRAM handshake
+// between the collector and the pipeline
+type SCRAMConfig struct {
+	userName           string `mapstructure:"userName"`
+	password           string `mapstructure:"password"`
+	algorithm          string `mapstructure:"algorithm"`
+	useTLS             bool   `mapstructure:"useTLS"`
+	insecureSkipVerify bool   `mapstructure:"insecureSkipVerify"`
+}
+
+func setSCRAMConfiguration(config *SCRAMConfig, saramaConfig *sarama.Config) error {
+	saramaConfig.Net.SASL.Enable = true
+	saramaConfig.Net.SASL.User = config.userName
+	saramaConfig.Net.SASL.Password = config.password
+
+	if config.algorithm == "sha512" {
+		saramaConfig.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &SCRAMClient{HashGeneratorFcn: SHA512} }
+		saramaConfig.Net.SASL.Mechanism = sarama.SASLMechanism(sarama.SASLTypeSCRAMSHA512)
+	} else if config.algorithm == "sha256" {
+		saramaConfig.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &SCRAMClient{HashGeneratorFcn: SHA256} }
+		saramaConfig.Net.SASL.Mechanism = sarama.SASLMechanism(sarama.SASLTypeSCRAMSHA256)
+	} else {
+		errors.Errorf("invalid SHA algorithm '%s': can be either 'sha256' or 'sha512'", config.algorithm)
```

Comment:
missing `return`? and consequently, missing unit test for it?

Reply:
I added a unit test with instantiating the scramConfig itself but adding initFromViper test soon if I can get it working.